### PR TITLE
Merge Dev Branch

### PR DESCRIPTION
**Pull request by momo-87**
### Added Features:
- Merge Dev Branch to main branch

### CODE REVIEW

File: src/styles/navbar.scss [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    // ul {
+    //   display: flex;
+    //   height: 60px;
+
+    //   li a {
+    //   color: #F1F6F9;
+    //   }
+    // }
```
Comment (zieeco) [inline src/styles/navbar.scss]: - Please, kindly remove these commented pieces of code to make your codebase clean and maintainable

File: src/components/OtherCitiesList.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import { useDispatch } from 'react-redux';
+import { NavLink } from 'react-router-dom';
+import { BsArrowRightCircle } from 'react-icons/bs';
+import PropTypes from 'prop-types';
+import { v4 as uuidv4 } from 'uuid';
+import { addClickedCity } from 'redux/home/homeSlice';
+
+const OtherCitiesList = ({ otherCities, isLoading, error }) => {
+  // const { otherCities, isLoading, error } = useSelector(getData);
```
Comment (zieeco) [inline src/components/OtherCitiesList.jsx]: - Please, kindly remove these commented pieces of code to make your codebase clean and maintainable

File: src/styles/navbar.scss (hunk 1 above)
Comment (zieeco) [reply src/styles/navbar.scss]: ## STYLES


- Hey, according to the project requirement, you should use the correct font family as in the design. Please, kindly improve on this below:

image

image

File: src/redux/home/homeSlice.js [hunk 3]
```diff
@@ -0,0 +1,512 @@
...
+            hu: weatherResp.data.main.humidity,
+            ws: weatherResp.data.wind.speed,
+            icon: weatherResp.data.weather[0].icon,
+          },
+        },
+        otherCities,
+      }
+    );
+  } catch (error) {
+    return rejectWithValue('Your Location is not supported !!');
```
Comment (zieeco) [inline src/redux/home/homeSlice.js:222]: **[OPTIONAL]**

- Hey, at the start of your page, it throws the error for some specific country, e.g Nigeria, the app would throw that error, The app should be accessible to any country of the world and as such, please, kindly improve on this to make the app accessible to any country to improve the user experience of your app.

File: src/redux/home/homeSlice.js [hunk 4]
```diff
@@ -0,0 +1,512 @@
...
```
Comment (zieeco) [inline src/redux/home/homeSlice.js:450]: **[OPTIONAL]**

- Hey, at the start of your page, it throws the error for some specific country, e.g Nigeria, the app would throw that error, The app should be accessible to any country of the world and as such, please, kindly improve on this to make the app accessible to any country to improve the user experience of your app.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+</details>
+<br>
+
+### Key Features <a name="key-features"></a>
+- A home page showing User city data and closest cities data
+- Details page showing details informations about a given location
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+>[Live Demo](https://airmetrics.onrender.com)
```
Comment (zieeco) [inline README.md:69]: ## VIDEO PRESENTATION

- Hey, great job adding the live demo version, however, you're yet to add the missing video presentation which is a key feature in this project to be approved. kindly see below the instructions to implement it below:

image

File: src/tests/App.test.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  it('will test App rendering', () => {
+    render(
+      <MemoryRouter>
+        <Provider store={store}>
+          <App />
+        </Provider>
+      </MemoryRouter>,
+    );
+  });
+});
```
Comment (zieeco) [inline src/tests/App.test.js:27]: -  [ ]  According to the project requirements :

   - The App should have unit tests for several components of the app and some pure functions in Redux code (currently you only tested the navbar component and App). The App should have integration tests using React Testing Library. Please, kindly add that 🙏

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @momo-87,





Great job so far!  👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Dice project implementation. ✔️
- Good commit messages :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Check the comments under the review._

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/styles/navbar.scss (hunk 1 above)
Comment (momo-87) [reply src/styles/navbar.scss]: Thank you, I really appreciate the time you spend reviewing my project.
Gill Sans and Lato font added to the project.

File: src/redux/home/homeSlice.js (hunk 4 above)
Comment (momo-87) [reply src/redux/home/homeSlice.js:450]: Unfortunately, the API provides the data so, I can manage this aspect. Thank you for your observation and feedback.

File: README.md (hunk 5 above)
Comment (momo-87) [reply README.md:69]: Thank you, I really appreciate the time you spend reviewing my project.
The video presentation link was added to the readme file.

File: src/tests/App.test.js (hunk 6 above)
Comment (momo-87) [reply src/tests/App.test.js:27]: Thank you, I really appreciate the time you spend reviewing my project.
I added two additional tests for SelectedCity and DetailsPages respectively.

File: src/tests/NearestCity.test.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import { render, screen } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import store from 'redux/store';
+import NearestCity from 'components/NearestCity';
+
+// eslint-disable-next-line react/prop-types
```
Comment (Ghiftee) [inline src/tests/NearestCity.test.js]: - [ ] Kindly do not disable linters as it helps us write cleaner code

File: src/tests/App.test.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+const store = configureStore({ reducer });
+describe('App test', () => {
+  it('will test App rendering', () => {
+    render(
+      <MemoryRouter>
+        <Provider store={store}>
+          <App />
+        </Provider>
+      </MemoryRouter>,
+    );
```
Comment (Ghiftee) [inline src/tests/App.test.js:25]: - [ OPTIONAL ] When I run the tests, I get a console error as shown below, I suggest that you consider fixing it. It is not enough for the tests to pass, but as good developers, we also need to pay attention to console errors.

Screenshot 2023-07-01 at 22 43 03

File: README.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+> [Video Presentation](https://www.loom.com/share/7d841e25e3ca4efd983da5fa5d90c2c2?sid=85dd4d9f-2734-4d6c-9233-d8c6a07ec87d)
```
Comment (Ghiftee) [inline README.md]: - [ ] Good job adding a video description of your project, however, you did not `highlight a piece of code you are proud of` as per the requirements shown below

Screenshot 2023-07-01 at 22 45 51 (2)

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hi there @momo-87 👋🏽 , 

Great work on making the changes requested by a previous reviewer 👏 
To highlight 👍🏽 
✔️ Tests are added
✔️ Tests pass
✔️ App runs as expected
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @momo-87, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**